### PR TITLE
Joomla 4.0: Removing JNode and JTree

### PR DESCRIPTION
**Pull request by Hackwar**
This PR removes JNode and JTree. Both classes have been broken basically since Joomla 1.5.

I'm actually a bit torn on this one. Yes, the current code is crap and should not be used, but at the same time I would love to have a proper interface for a tree structure and its nodes. It would be nice to have the categories and menu system use such a class and maybe even the user groups. However that seems to be out of the scope for 4.0 and might even be a performance issue.

Long story short: Scrap this and be done with it.

Oh, and I took the liberty of removing the reference to JSimpleCrypt.

### CODE REVIEW

**Comment by wilsonge** [comment]
Docs removed without replacement https://docs.joomla.org/Potential_backward_compatibility_issues_in_Joomla_4#Classes_Removed_Without_Replacement
